### PR TITLE
LIVE-3218 prevent resources to go undefined

### PR DESCRIPTION
**Pull request by henrily-ledger**
### 📝 Description

CoinResources happened to be undefined and cause exception because it's set by undefined by default

### ❓ Context

- **Impacted projects**: `live-common` 
- **Linked resource(s)**: `https://ledgerhq.atlassian.net/browse/LIVE-3218` 

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

**Review by nparigi-ledger — APPROVED** [review]

**Review by hedi-edelbloute — APPROVED** [review]

**Review by Justkant — DISMISSED** [review]
I just saw that we probably need a bit more checks on some variables

File: libs/ledger-live-common/src/account/helpers.ts [hunk 1]
```diff
@@ -247,9 +247,11 @@ export const getVotesCount = (
     case "tezos":
       return isAccountDelegating(account) ? 1 : 0;
     case "tron":
-      return (mainAccount as TronAccount).tronResources.votes.length;
+      return (mainAccount as TronAccount)?.tronResources.votes.length || 0;
```
[the remark this answers is not in the document]
Comment (henrily-ledger) [reply libs/ledger-live-common/src/account/helpers.ts:250]: Nice, my bad get the wrong optionnal thanks !

**Review by hedi-edelbloute — APPROVED** [review]

**Comment by henrily-ledger** [comment]
@Justkant is it possible to review this PR again with your suggested change ?

File: libs/ledger-live-common/src/account/serialization.ts [hunk 2]
```diff
@@ -818,26 +827,37 @@ export function fromAccountRaw(rawAccount: AccountRaw): Account {
...
+    }
+    case "crypto_org": {
+      const cryptoOrgResourcesRaw = (rawAccount as CryptoOrgAccountRaw)
+        .cryptoOrgResources;
       (res as CryptoOrgAccount).cryptoOrgResources = fromCryptoOrgResourcesRaw(
-        (rawAccount as CryptoOrgAccountRaw).cryptoOrgResources
+        cryptoOrgResourcesRaw
       );
       break;
+    }
```
Comment (haammar-ledger) [inline libs/ledger-live-common/src/account/serialization.ts:860]: Why not check `if (xxxRawResources)` for Bitcoin, Cardano, Solana and CryptoOrg ? Same as you did for the others.

Comment (henrily-ledger) [reply libs/ledger-live-common/src/account/serialization.ts:860]: oops will do thx i forgot about this one

**Review by Justkant — APPROVED** [review]

**Review by haammar-ledger — APPROVED** [review]

**Pull request merged**